File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/presentation/wrappers/AndroidResourcesWrapper.java
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.wrappers;

import android.util.Patterns;


public class AndroidResourcesWrapper {

    public boolean getResource(String email) {
        return Patterns.EMAIL_ADDRESS.matcher(email).matches();
    }
}

File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/presentation/wrappers/AndroidResourcesWrapper.kt
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.wrappers

import android.util.Patterns

class AndroidResourcesWrapper {

    fun getResource(email: String): Boolean {
        return Patterns.EMAIL_ADDRESS.matcher(email).matches()
    }
}
